src/ovi/Provider.php

```diff
@@ -0,0 +1,468 @@
+
+           if($username_valid ==  'false')
+           {
+        	$error = 'Account Terminate Failed. Username '.$username.' does not exists.';
+		return $this->errorResult('Error: ' . $error);
+           }
+
+            $input_tag='<username>'.$username.'</username>';
+            $return = $this->ovipanel_CurlExecute( $input_tag, "DeleteClient");
+	    $data = json_decode($return);
+		if ($data === null) {
+	        $error = "Failed to decode JSON";
+        	return $this->errorResult('Error: ' . $error);
+            	}
+	         elseif ($data->success) {
+          	return $this->emptyResult('Account deleted');
+		}
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function grantReseller(GrantResellerParams $params): ResellerPrivileges
+    {
+        throw $this->errorResult('Not implemented');
+           $username = runtime_sanatizeItem::sanatizeItem(trim($params->username),"string");
+           $username_valid = $this->ovipanel_UsernameExists($params->username);
+
+           if($username_valid ==  'false')
+           {
+        $error = 'Account grantReseller Failed. Username '.$username.' does not exists.';
+           return $this->errorResult('Error: ' . $error);
+	   }
+
+            $input_tag='<username>'.$username.'</username>';
+            $return = $this->ovipanel_CurlExecute( $input_tag, "grantReseller");
+
+            $data = json_decode($return);
+                if ($data === null) {
+           $error = "Failed to decode JSON";
+           return $this->errorResult('Error: ' . $error);
+            }
+         elseif ($data->success) {
+          return ResellerPrivileges::create()
+            ->setMessage('Reseller privileges granted')
+            ->setReseller(true);
+	   } 
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function revokeReseller(AccountUsername $params): ResellerPrivileges
+    {
+        throw $this->errorResult('Not implemented');
+
+           $username = runtime_sanatizeItem::sanatizeItem(trim($params->username),"string");
+           $username_valid = $this->ovipanel_UsernameExists($params->username);
+
+           if($username_valid ==  'false')
+           {
+        	$error = 'Account revokeReseller Failed. Username '.$username.' does not exists.';
+	        return $this->errorResult('Error: ' . $error);
+           }
+
+            $input_tag='<username>'.$username.'</username>';
+            $return = $this->ovipanel_CurlExecute( $input_tag, "revokeReseller");
+
+            $data = json_decode($return);
+                if ($data === null) 
+		{
+		        $error = "Failed to decode JSON";
+		        return $this->errorResult('Error: ' . $error);
+            	}
+	         elseif ($data->success) {
+		   return ResellerPrivileges::create()->setMessage('Reseller privileges revoked')->setReseller(false);
+          	}
+    }
+
+    /**
+     * @inheritDoc
+     */
+
+      public function sanatizeItem($var, $type)
```

Comment:
FYI for escaping inputs to be concatenated into XML you can often simply use `htmlspecialchars()`